components/AudioPlayer.tsx

```diff
@@ -0,0 +1,388 @@
+import { Box, IconButton } from "@mui/material";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const AudioPlayer = () => {
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+  const [activeVolume, setActiveVolume] = useState(false);
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  const [volume, setVolume] = useState(100);
```

Comment:
idk, maybe all the audio stuff should be inside a single object, to isolate this audio state?

Reply:
Hmm, I think it's good to encapsulate all the state variables into a single object

Reply:
I leave this idea because when we encapsulate variables into a single object, in useEffect hooks we need to add in dependency this object "audioState" and take useless actuations of useEffect and take many strange behaviors...

So I leave this idea, because I looked at the code and see the I get more rows with this encapsulating all the state variables into a single object than before and many bugs took with this encapsulating